volatility/framework/symbols/windows/pdb.py

```diff
@@ -0,0 +1,288 @@
+# This file is Copyright 2020 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import binascii
+import json
+import logging
+import lzma
+import os
+import struct
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
+from urllib import request
+
+from volatility import symbols
+from volatility.framework import constants, interfaces
+from volatility.framework.configuration.requirements import SymbolTableRequirement
+from volatility.framework.symbols import intermed
+from volatility.framework.symbols.windows import pdbconv
+
+vollog = logging.getLogger(__name__)
+
+
+class PDBUtility:
+    """Class to handle and manage all getting symbols based on MZ header"""
+
+    @classmethod
+    def symbol_table_from_offset(cls,
+                                 context: interfaces.context.ContextInterface,
+                                 layer_name: str,
+                                 offset: int,
+                                 symbol_table_class: str,
+                                 config_path: str = None,
+                                 progress_callback: constants.ProgressCallback = None) -> str:
+        """Produces the name of a symbol table loaded from the offset for an MZ header
+
+        Args:
+            context: The context on which to operate
+            layer_name: The name of the (contiguous) layer within the context that contains the MZ file
+            offset: The offset in the layer at which the MZ file begins
+            symbol_table_class: The type of symbol class to construct the SymbolTable
+            config_path: New path for the produced symbol table configuration with the config tree
+            progress_callback: Callable called to update ongoing progress
+
+        Returns:
+            None if no pdb information can be determined, else returned the name of the loaded symbols for the MZ
+        """
+        guid, age, pdb_name = cls.get_guid_from_mz(context, layer_name, offset)
```

Comment:
`get_guid_from_mz` can return `None`, so I think this is not safe. It could cause an exception in framework code.

Reply:
Good call, should now be fixed.  5:)